unidist/core/backends/mpi/core/communication.py

```diff
@@ -53,6 +54,35 @@ def get_mpi_state():
     return comm, rank, world_size
 
 
+def get_topology(is_call_from_worker=False):
```

Comment:
Looking forward to what we need (ranks map) I would suggest to have a unified topology object on each rank/process with the following format - `{"node-ip" : {"CPU" : 1, "Ranks" : [ranks list] }`
Therefore, ``allgather`` should get a host-ip and rank, that is all.

Reply:
This is implemented. The format you proposed for `topology` variable is changed to `{"node_ip": worker_ranks_list, ..}`. No specific need in `CPU` key here, because it always can be computed as `len(ranks_list)`